<?xml version="1.0"?>
<launch>
  <arg name="paused" default="false"/>
  <arg name="gazebo_gui" default="true"/>
  <arg name="urdf_path" default="$(find gazebocontrolsim)/urdf/gazebo_mutifingerhand.urdf"/>

  <!-- startup simulated world -->
  <include file="$(find gazebocontrolsim)/launch/hit_world.launch">
    <arg name="world_name" default="worlds/empty.world"/>
    <arg name="paused" value="$(arg paused)"/>
    <arg name="gui" value="$(arg gazebo_gui)"/>
  </include>

  <!-- send robot urdf to param server -->
  <param name="robot_description" textfile="$(arg urdf_path)" />

  <!-- push robot_description to factory and spawn robot in gazebo at the origin, change x,y,z arguments to spawn in a different position -->
  <node name="spawn_gazebo_model" pkg="gazebo_ros" type="spawn_model" args="-urdf -param robot_description -model mutifingerhand -x 0 -y 0 -z 0"
    respawn="false" output="screen" />

  <!-- Load joint controller configurations from YAML file to parameter server -->
  <rosparam file="$(find gazebocontrolsim)/config/ros_controllers.yaml" command="load"/>

  <!-- Load the controllers -->
  <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
    output="screen" ns="/mutifingerhand" args="J11_controller
    J12_controller
    J13_controller
    J14_controller
    J15_controller
    J21_controller
    J22_controller
    J23_controller
    J24_controller
    J31_controller
    J32_controller
    J33_controller
    J34_controller
    J41_controller
    J42_controller
    J43_controller
    J44_controller
    J51_controller
    J52_controller
    J53_controller
    J54_controller"/>

  <!--  Publish the desired joint_states-->
  <node name="pose_control" pkg="gazebocontrolsim" type="pose_control" output="screen">
  </node>

  <!-- convert joint states to TF transforms for rviz, etc -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"
    respawn="false" output="screen">
    <remap from="/joint_states" to="/mutifingerhand/joint_states" />
  </node>

</launch>
